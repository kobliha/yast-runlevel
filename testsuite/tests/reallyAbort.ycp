// test for ui.ycp::reallyAbort

{
    include "testsuite.ycp";
    include "runlevel/ui.ycp";
    include "runlevel/services.ycp";

    UI (``{
	global define AnyQuestionPopup (any a, any b, any c, any d, any e) ``{
	    y2error ("AnyQuestionPopup returning true");
	    return true;
	}
    });

    string default_runlevel = "0";
    string default_runlevel_orig = "0";
    map services = $[];
    boolean do_abort_now = false;
    reallyAbort ();
    DUMP (do_abort_now);

    default_runlevel = "1";
    default_runlevel_orig = "0";
    services = $[];
    do_abort_now = false;
    reallyAbort ();
    DUMP (do_abort_now);

    default_runlevel = "1";
    default_runlevel_orig = "1";
    services = $[ "usbmgr" : $[ "dirty" : true ] ];
    do_abort_now = false;
    reallyAbort ();
    DUMP (do_abort_now);

    default_runlevel = "0";
    default_runlevel_orig = "1";
    services = $[ "usbmgr" : $[ "dirty" : true ] ];
    do_abort_now = false;
    reallyAbort ();
    DUMP (do_abort_now);

    do_abort_now = true;
    reallyAbort ();
    DUMP (do_abort_now);
}
