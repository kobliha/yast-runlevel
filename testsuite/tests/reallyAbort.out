Dump	true
Log	AnyQuestionPopup returning true
Dump	true
Log	AnyQuestionPopup returning true
Dump	true
Log	AnyQuestionPopup returning true
Dump	true
Dump	true
